tests/Unit/GeneralAssemblyTest.php

```diff
@@ -166,4 +167,28 @@ public function test_passive_users_get_excused_automatically(): void
         $this->assertTrue($excused->contains($userPassive1) && $excused->contains($userPassive2));
         $this->assertNotNull($excused->first()->pivot->comment); // Check if excuse reason is set
     }
+
+    /**
+     * @return void
+     */
+    public function test_new_students_pass_requirements(): void
+    {
+        $user = User::factory()->create(['verified' => true]);
+        $user->educationalInformation()->save(EducationalInformation::factory()->make());
+        $user->educationalInformation->update(['year_of_acceptance' => now()->year]);
+
+        $generalAssembly = GeneralAssembly::factory()->create();
+        $generalAssembly2 = GeneralAssembly::factory()->create();
+
+        $generalAssembly->update(['closed_at' => now()->setYear(now()->year)->setMonth(2)->setDay(15)]);
+        $generalAssembly2->update(['closed_at' => now()->setYear(now()->year)->setMonth(9)->setDay(15)]);
+
+        $generalAssembly->presenceChecks()->create();
```

Comment:
Technically, the user should pass even without presenceChecks, shouldn't? Or if you need them, please add a comment why.

Reply:
Before the modification to `GeneralAssembly.php/requirementsPassed`, the user would have failed the assert after the presenceChecks, because they didn't take part in the last two general assemblies. However, note that the second last general assembly in the test (02.15) is before the acceptance date of the user (09.01), so the user shouldn't be expected to have taken part in that GA, thus they should pass the requirements.

Reply:
Fun fact: if the number of presence checks during a GA is 2, the users are expected to take part in both of them. However, when the number of presence checks is 3, the users are expected to take part in only 1, cf.
https://github.com/EotvosCollegium/mars/blob/cedc643006735db86b395d522f91866f97687d23/app/Models/GeneralAssemblies/GeneralAssembly.php#L199-L203
I don't know what would be the expected behavior here, should I open an issue regarding this?

Reply:
I think you can simply use `$presenceCheckCount <= 2 ? $presenceCheckCount : max($presenceCheckCount - 2, 2)`.

Reply:
Just make sure to update any description for the users, if we state the calculation method anywhere.